ui/src/app/navigation/libraryexport.component.ts

```diff
@@ -0,0 +1,41 @@
+import {Component, OnInit, LOCALE_ID, Inject} from "@angular/core"
+import {formatDate} from "@angular/common"
+import {SearchService} from "../search/search.service"
+import {RichSkillService} from "../richskill/service/rich-skill.service"
+import {ActivatedRoute} from "@angular/router"
+import {AuthService} from "../auth/auth-service"
+import * as FileSaver from "file-saver"
+import {SvgHelper, SvgIcon} from "../core/SvgHelper"
+import {CommoncontrolsComponent} from "./commoncontrols.component"
+import {AbstractSearchComponent} from "./abstract-search.component"
+
+@Component({
+  selector: "app-libraryexport",
+  templateUrl: "./libraryexport.component.html"
+})
+export class LibraryExportComponent extends AbstractSearchComponent implements OnInit {
+
+  searchIcon = SvgHelper.path(SvgIcon.SEARCH)
+  dismissIcon = SvgHelper.path(SvgIcon.DISMISS)
+
+  constructor(
+    protected searchService: SearchService,
+    protected route: ActivatedRoute,
+    protected authService: AuthService,
+    protected richSkillService: RichSkillService,
+    @Inject(LOCALE_ID) protected locale: string) {
+    super(searchService, route, authService)
+  }
+
+  ngOnInit(): void {
+  }
+
+  onDownloadLibrary(): void {
+    this.richSkillService.libraryExport()
+      .subscribe((csv: string) => {
+        const blob = new Blob([csv], {type: "text/csv;charset=utf-8;"})
+        const date = formatDate(new Date(), "yyyy-MM-dd", this.locale)
+        FileSaver.saveAs(blob, `RSD Library - OSMT ${date}.csv`)
```

Comment:
@wgu-edwin, I'd like to see this covered by unit test as well. Something along the lines of

```
    spyOn(FileSaver, 'saveAs').and.stub();
```
Looking for saveAs to be called with a Blob of type CSV and a file name matching most of the template string in the code (startsWith and endsWith. Don't get hung up on the date, that's enough coverage to prevent accidental drift in the future).

Reply:
Per our verbal, we are going to punt on fine-grained testing of the arguments for now, and upgrade to the latest Jasmine, which makes that easy.